modules/datagrid/72/launch/added/launch/infinispan-config.sh

```diff
@@ -282,16 +282,47 @@ function configure_cache() {
        </encoding>"
   fi
 
-  if [ -n "$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")" ]; then
-    if [ -n "$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")" ]; then
-      local CACHE_EVICTION_STRATEGY="strategy=\"$(find_env "${prefix}_CACHE_EVICTION_STRATEGY")\""
-    fi
+  if [ -n "$(find_env "${prefix}_CACHE_MEMORY_EVICTION_SIZE")$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")" ]; then
+
+    local evictionSize="$(find_env "${prefix}_CACHE_MEMORY_EVICTION_SIZE")"
     if [ -n "$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")" ]; then
-      local CACHE_EVICTION_MAX_ENTRIES="size=\"$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")\""
+      log_warning "Environment variable ${prefix}_CACHE_EVICTION_MAX_ENTRIES has been deprecated. Use ${prefix}_CACHE_MEMORY_EVICTION_SIZE instead."
+      if [ -z "$(find_env "${prefix}_CACHE_MEMORY_EVICTION_SIZE")" ]; then
+      	evictionSize="$(find_env "${prefix}_CACHE_EVICTION_MAX_ENTRIES")"
+      fi
+    fi
+
+    #if cache eviction type is not specified default is COUNT
+    local evictionType="COUNT"
+    if [ "$(find_env "${prefix}_CACHE_MEMORY_EVICTION_TYPE")" == "MEMORY" ]; then
+      evictionType="MEMORY"
+    fi
+
+
+    #if cache eviction strategy is not specified default is REMOVE
+    local evictionStrategyType="REMOVE"
+    if [ -n "$(find_env "${prefix}_CACHE_MEMORY_EVICTION_STRATEGY")" ]; then
+      evictionStrategyType="$(find_env "${prefix}_CACHE_MEMORY_EVICTION_STRATEGY")"
```

Comment:
Unfortunately if strategy is EXCEPTION then we have to make the cache transactional and have at least NON_DURABLE_XA